torch/_inductor/codegen/triton_foreach.py

```diff
@@ -123,14 +184,26 @@ def codegen_kernel(self, name=None):
         code.writeline(f"def {name or 'KERNEL_NAME'}({', '.join(argdefs)}):")
 
         with code.indent():
-            code.splice("pid = tl.program_id(0)")
-            code.splice(f"XBLOCK: tl.constexpr = {self.block_size}")
+            code.splice("xpid = tl.program_id(0)")
+            if self.blocking_2d:
+                code.splice("ypid = tl.program_id(1)")
+                code.splice(f"XBLOCK: tl.constexpr = {self.block_size_2d}")
+                code.splice(f"YBLOCK: tl.constexpr = {self.block_size_2d}")
+            else:
+                code.splice(f"XBLOCK: tl.constexpr = {self.block_size_1d}")
 
             for sub_kernel in self.sub_kernels:
-                num_elems = int(sympy_product(sub_kernel.numels))
-                self.codegen_pid_range(code, num_elems)
+                assert len(sub_kernel.numels) <= 3
+                # TODO mlazos: support dynamic shapes
+                numel_ind = 0 if not self.blocking_2d else 1
+                self.codegen_pid_range(code, int(sub_kernel.numels[numel_ind]))
                 with code.indent():
-                    code.splice(f"xnumel = {num_elems}")
+                    if self.blocking_2d:
+                        code.splice(f"ynumel = {sub_kernel.numels[0]}")
+                        code.splice(f"xnumel = {sub_kernel.numels[1]}")
```

Comment:
This interpretation of x/y pids is consistent with https://github.com/pytorch/pytorch/pull/107902 . cc @jansel